### PR TITLE
add tests for support of xx-YY locales in embed/public-links

### PR DESCRIPTION
**Pull request by npretto**
This PR changes some e2e tests so that they test the fallback logic added in https://github.com/metabase/metabase/pull/51002

### CODE REVIEW

File: e2e/test/scenarios/embedding/embedding-dashboard.cy.spec.js [hunk 1]
```diff
@@ -1002,11 +1002,16 @@ H.describeEE("scenarios > embedding > dashboard appearance", () => {
       },
       {
         additionalHashOptions: {
-          locale: "de",
+          locale: "de-CH",
```
Comment (heypoom) [inline e2e/test/scenarios/embedding/embedding-dashboard.cy.spec.js:1005]: I wonder if we have separate tests for languages codes (`de`) and language-country codes (`de-CH`) somewhere else in the codebase? Might be good to test both cases

File: e2e/test/scenarios/embedding/embedding-dashboard.cy.spec.js [hunk 2]
```diff
@@ -1002,11 +1002,16 @@ H.describeEE("scenarios > embedding > dashboard appearance", () => {
...
         },
       },
     );
 
+    // We don't have a de-CH.json file, so it should fallback to de.json, see metabase#51039 for more details
+    cy.request("/app/locales/de.json").then(response => {
```
Comment (heypoom) [inline e2e/test/scenarios/embedding/embedding-dashboard.cy.spec.js:1011]: Should this be `cy.intercept` instead? `cy.request` makes the request while `cy.intercept` checks if the app makes that request afaik

**Review by heypoom — APPROVED** [review]

File: e2e/test/scenarios/embedding/embedding-dashboard.cy.spec.js (hunk 1 above)
Comment (npretto) [reply e2e/test/scenarios/embedding/embedding-dashboard.cy.spec.js:1005]: https://github.com/metabase/metabase/pull/51002#discussion_r1884230079
I added a lot more unit tests in that PR to test the underlying logic, this is just a check that the fallback logic is indeed used here

Comment (heypoom) [reply e2e/test/scenarios/embedding/embedding-dashboard.cy.spec.js:1005]: Perfect, thanks!

File: e2e/test/scenarios/embedding/embedding-dashboard.cy.spec.js (hunk 2 above)
Comment (npretto) [reply e2e/test/scenarios/embedding/embedding-dashboard.cy.spec.js:1011]: Yes! Thanks for spotting that 🙏

Comment (npretto) [reply e2e/test/scenarios/embedding/embedding-dashboard.cy.spec.js]: 106717f8c3715e347ae8c5230597f493ec499aba

**Review by WiNloSt — APPROVED** [review]

**Pull request merged**